osu.Game/Scoring/ScoreInfo.cs

```diff
@@ -319,6 +333,17 @@ public IEnumerable<HitResultDisplayStatistic> GetStatisticsForDisplay()
 
         public bool Equals(ScoreInfo? other) => other?.ID == ID;
 
+        protected override void OnPropertyChanged(string propertyName)
+        {
+            base.OnPropertyChanged(propertyName);
+
+            //Provides hash update
+            if (propertyName == nameof(BeatmapInfo))
+            {
+                BeatmapHash = BeatmapInfo.Hash;
+            }
+        }
```

Comment:
Conceptually and from a source level, I like this - but then I started reading realm docs on this and got a bit worried if the semantics of this callback are going to allow us to safely use it for this use case. Read https://github.com/realm/realm-dotnet/discussions/3230 for more.

Reply:
I also wrote this test to try and check experimentally but it's failing (not in a way I'd expect it to fail at least) and I have no clue why. It looks like the change callback fires twice, but not the time I expect it to (which is at the explicit `BeatmapInfo` assignment):

```csharp
        [Test]
        public void TestHashTransfer()
        {
            RunTestWithRealm((realm, _) =>
            {
                BeatmapSetInfo beatmapSet = null!;
                ScoreInfo score = null!;
                string hashDuringTransaction = string.Empty;

                realm.Write(r =>
                {
                    beatmapSet = TestResources.CreateTestBeatmapSetInfo();
                    r.Add(beatmapSet);

                    score = TestResources.CreateTestScoreInfo(r.All<RulesetInfo>().First());
                    r.Add(score);
                    score.BeatmapInfo = beatmapSet.Beatmaps.First();

                    hashDuringTransaction = score.BeatmapHash;
                });

                Assert.Multiple(() =>
                {
                    Assert.That(hashDuringTransaction, Is.EqualTo(beatmapSet.Beatmaps.First().Hash));
                    Assert.That(score.BeatmapHash, Is.EqualTo(beatmapSet.Beatmaps.First().Hash));
                });
            });
        }
```

Place in `GeneralUsageTests` to run.

Reply:
Appears that the realm team's answer as to whether this is valid usage is [no](https://github.com/realm/realm-dotnet/discussions/3230#discussioncomment-4976791), so unfortunately I don't think we can proceed with this as is.

That said I will probably yield and stop blocking this PR if the version from https://github.com/ppy/osu/commit/4fdba880b1efc2dcb49a2f3c64585f2e896a64ba is brought back (so strictly manual handling of the beatmap hash on the score). All of the variants that attempted to manage it automatically are just either too weird or too broken to live.

Reply:
For now I have no idea other than adding class that provides this update (which is not what you want I guess). Since we can't come up with anything better, I'm ok with 4fdba88.

Reply:
I'd propose going back to 4fdba88 for now as I'm still generally okay with that approach.

Would require @bdach to not be strongly against it though.

Reply:
I can _tolerate_ 4fdba88. We'll see how many times this rake is going to be stepped on (maybe never?)